### PR TITLE
Upgrade dependencies and Reflect recent observatorium/deployment changes

### PR DESCRIPTION
**Pull request by kakkoyun**
Signed-off-by: Kemal Akkoyun <kakkoyun@gmail.com>

depends xref: https://github.com/observatorium/deployments/pull/368 and https://github.com/observatorium/deployments/pull/376

### CODE REVIEW

File: example/main.jsonnet [hunk 1]
```diff
@@ -1,40 +1,38 @@
...
-
-local obs = (import 'github.com/observatorium/deployments/environments/base/observatorium.jsonnet');
+  version: 'v2.24.0',
+  image: 'quay.io/dexidp/dex:v2.24.0',
+  commonLabels+:: {
+    'app.kubernetes.io/instance': 'e2e-test',
+  },
+});
 
+local obs = (import 'github.com/observatorium/deployments/components/observatorium.libsonnet');
```
Comment (metalmatze) [inline example/main.jsonnet:35]: I'd really move all imports to the top. 
Hasn't been that way before, but that's generally what we try to do - making it simliar to Go and now have imports all over the place.

Comment (metalmatze) [reply example/main.jsonnet:35]: Obviously can be done in another PR or never if not deemed important enough.

File: jsonnet/obs-operator.jsonnet [hunk 2]
```diff
@@ -1,82 +1,131 @@
...
 local loki = import 'github.com/observatorium/deployments/components/loki.libsonnet';
-local config = import './operator-config.libsonnet';
-local obs = ((import 'github.com/observatorium/deployments/components/observatorium.libsonnet') + {
-               config+:: config,
-             } + (import 'github.com/observatorium/deployments/components/observatorium-configure.libsonnet'));
+local memcached = import 'github.com/observatorium/deployments/components/memcached.libsonnet';
+local api = import 'github.com/observatorium/observatorium/jsonnet/lib/observatorium-api.libsonnet';
+local receiveController = import 'github.com/observatorium/thanos-receive-controller/jsonnet/lib/thanos-receive-controller.libsonnet';
+local thanos = import 'github.com/thanos-io/kube-thanos/jsonnet/kube-thanos/thanos.libsonnet';
+local obs = (import 'github.com/observatorium/deployments/components/observatorium.libsonnet');
```
Comment (metalmatze) [inline jsonnet/obs-operator.jsonnet:7]: I like seeing this. :+1:

**Review by metalmatze — COMMENTED** [review]
Nothing really blocking just some comments on optional improvements. 
Nice job! :100:

**Comment by kakkoyun** [comment]
cc @clyang82

**Comment by clyang82** [comment]
Thanks @kakkoyun 💯

**Pull request merged**